src/lib/mediator.ts

```diff
@@ -0,0 +1,60 @@
+import { MediatorEventOptions } from "typings/mediator"
+import EventEmitter from "eventemitter3"
+
+export interface Mediator {
+  emitter: any
+  trigger: (action: string, options?: MediatorEventOptions) => void
+  on: (event: string, cb?: (payload?: MediatorEventOptions) => void) => void
+  off: (event: string) => void
+  once: (event: string, cb?: (payload?: MediatorEventOptions) => void) => void
+}
+
+declare global {
+  interface Window {
+    __mediator: Mediator["emitter"]
+  }
+}
+
+const emitter: Mediator["emitter"] =
+  typeof window !== "undefined"
+    ? window.__mediator || (window.__mediator = new EventEmitter())
+    : new EventEmitter()
+
+const trigger: Mediator["trigger"] = (
+  eventName: string,
+  options?: MediatorEventOptions,
+  optionalData?: object
+) => {
+  emitter.emit(eventName, options, optionalData)
+}
+
+const on: Mediator["on"] = (
+  eventName: string,
+  callback: (options?: MediatorEventOptions, optionalData?: object) => void
+) => {
+  emitter.on(eventName, callback)
+}
+
+const off: Mediator["off"] = (eventName: string) => {
+  emitter.off(eventName)
+}
+
+const once: Mediator["once"] = (
+  eventName: string,
+  callback: (options?: MediatorEventOptions) => void
+) => {
+  emitter.once(eventName, callback)
+}
+
+/**
+ * Event emitter used for (mostly) modal-related events
+ * @param eventName name of event track
+ * @param callback function to be called when named event is triggered
+ */
+export const mediator: Mediator = {
+  emitter,
```

Comment:
Does it make sense to make this to a class? It would remove the need for the extra interface definition and this simulates a class.

Reply:
We are extending the `EventEmitter` interface with `trigger`, which is our most widely used feature for `mediator`-- would rather keep the export as an object for now, but updated the typings to be more specific.

Reply:
Could you elaborate a little on what you mean by extending `EventEmitter`? I understand that adding these provides us with a drop-in replacement for the current mediator however I wouldn't consider this an extension of the EventEmitter.

One way that I feel like this would benefit from being a class is that with constructor support it would be really easy to injecting the emitter as a dependency at runtime, thus making testing possible without the need for mocks.